src/libsync/updatefiledropmetadata.cpp

```diff
@@ -0,0 +1,194 @@
+    connect(fetchMetadataJob, &GetMetadataApiJob::jsonReceived, this, &UpdateFileDropMetadataJob::slotFolderEncryptedMetadataReceived);
+    connect(fetchMetadataJob, &GetMetadataApiJob::error, this, &UpdateFileDropMetadataJob::slotFolderEncryptedMetadataError);
+
+    fetchMetadataJob->start();
+}
+
+void UpdateFileDropMetadataJob::slotFolderEncryptedMetadataError(const QByteArray &fileId, int httpReturnCode)
+{
+    Q_UNUSED(fileId);
+    Q_UNUSED(httpReturnCode);
+    qCDebug(lcUpdateFileDropMetadataJob()) << "Error Getting the encrypted metadata. Pretend we got empty metadata.";
+    const FolderMetadata emptyMetadata(propagator()->account());
+    const auto encryptedMetadataJson = QJsonDocument::fromJson(emptyMetadata.encryptedMetadata());
+    slotFolderEncryptedMetadataReceived(encryptedMetadataJson, httpReturnCode);
+}
+
+void UpdateFileDropMetadataJob::slotFolderEncryptedMetadataReceived(const QJsonDocument &json, int statusCode)
+{
+    qCDebug(lcUpdateFileDropMetadataJob) << "Metadata Received, Preparing it for the new file." << json.toVariant();
+
+    // Encrypt File!
+    _metadata = new FolderMetadata(propagator()->account(), json.toJson(QJsonDocument::Compact), statusCode);
+    if (!_metadata->moveFileDropToFiles()) {
+        unlockFolder();
+        return;
+    }
+
+    const auto updateMetadataJob = new UpdateMetadataApiJob(propagator()->account(), _folderId, _metadata->encryptedMetadata(), _folderToken);
+    connect(updateMetadataJob, &UpdateMetadataApiJob::success, this, &UpdateFileDropMetadataJob::slotUpdateMetadataSuccess);
+    connect(updateMetadataJob, &UpdateMetadataApiJob::error, this, &UpdateFileDropMetadataJob::slotUpdateMetadataError);
+    updateMetadataJob->start();
+}
+
+void UpdateFileDropMetadataJob::slotUpdateMetadataSuccess(const QByteArray &fileId)
+{
+    Q_UNUSED(fileId);
+    qCDebug(lcUpdateFileDropMetadataJob) << "Uploading of the metadata success, Encrypting the file";
+
+    qCDebug(lcUpdateFileDropMetadataJob) << "Finalizing the upload part, now the actuall uploader will take over";
+    unlockFolder();
+}
+
+void UpdateFileDropMetadataJob::slotUpdateMetadataError(const QByteArray &fileId, int httpErrorResponse)
+{
+    qCDebug(lcUpdateFileDropMetadataJob) << "Update metadata error for folder" << fileId << "with error" << httpErrorResponse;
+    qCDebug(lcUpdateFileDropMetadataJob()) << "Unlocking the folder.";
+    unlockFolder();
+}
+
+void UpdateFileDropMetadataJob::slotFolderLockedError(const QByteArray &fileId, int httpErrorCode)
+{
+    Q_UNUSED(httpErrorCode);
+    qCDebug(lcUpdateFileDropMetadataJob) << "Folder" << fileId << "Coundn't be locked.";
+    emit finished(SyncFileItem::Status::NormalError);
+}
+
+void UpdateFileDropMetadataJob::slotFolderEncryptedIdError(QNetworkReply *r)
+{
+    Q_UNUSED(r);
+    qCDebug(lcUpdateFileDropMetadataJob) << "Error retrieving the Id of the encrypted folder.";
+}
+
+void UpdateFileDropMetadataJob::unlockFolder()
+{
+    ASSERT(!_isUnlockRunning);
```

Comment:
are you sure you did not wanted to use `Q_ASSERT` ?
this `ASSERT` macro is also doing something in release mode